packages/blade/src/components/Switch/_decisions/decisions.md

```diff
@@ -0,0 +1,85 @@
+# Switch API Decisions <!-- omit in toc -->
+
+A switch component is used to quickly switch between two possible states. These are only used for binary actions that occur immediately after the user turn the switch on/off.
+
+<img src="./switch-thumbnail.png" width="380" alt="" />
+
+- [Design](#design)
+- [`Switch` API](#switch-api)
+  - [Examples:](#examples)
+    - [Basic](#basic)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=31366-831710&t=qzTLJeIDluL4Ps23-0) to all variants of the Switch component
+
+
+## `Switch` API
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| isChecked | `boolean` | `false` | State of the switch, if explicitly set the switch will become [controlled component](https://reactjs.org/docs/forms.html#controlled-components) |        |
+| defaultChecked | `boolean` | `false` | Default state of the switch, if isChecked is not provided the switch will become [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html) |        |
+| onChange | `({ event, isChecked, value }) => void` | `undefined` | The function to be called when the switch state changes. |        |
+| isDisabled | `boolean` | `false` | Control whether the switch is disabled or not. |        |
+| accessibilityLabel | `string` | `undefined` | Provides accessible label for internal checkbox component that sets the aria-label prop for screen readers. |        |
+| size | `small, medium` | `medium` | Size of the switch component |        |
+| name (html) | `string` | `undefined` | The name of the input field in a switch, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |        |
+| value (html) | `string` | `undefined` | The value of the input field in a switch, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value) |        |
+| id (html) | `string` | `undefined` | The id of the input field in a switch, useful for associating a label element with the input via `htmlFor` prop |        |
```

Comment:
can you create an issue to see if this can be added to other inputs as well? seems useful. We can discuss over the issue

Reply:
You are talking about the ID prop right? 

Actually it might just only make sense for Switch input only since this is the only component where we are not explicitly exposing label <Swtich label="Toggle darkmode" \/>, thats why we need the ID to associate with a htmlFor label. 

In other inputs we already have built-in labels, `<Checkbox>Toggle darkmode</Checkbox>`, `<TextInput label="" />` etc

Reply:
We haven't intentionally exposed `id` on any other components btw

Reply:
oh got it. makes sense.

Reply:
yes but there's an ask where we will have to make labels optional(bunch of places on dashboards) so I would say lets not expose the id yet until we figure out all the solution for all form elements? Anyways it'll be a broken functionality for form submissions but that's an edge case and we have ariaLabel for a11y so safe to take this tradeoff(yes it's broken functionality hence tradeoff)

Reply:
If switches aren't meant to be used as form fields (hence from design side no labels support), perhaps we don't need `id` support right now

Reply:
> If switches aren't meant to be used as form fields 

Okay then let's take it as an enhancement later then.

Reply:
+1 to @divyanshu013. Anyways we can keep value and other prop but can probably remove the id prop. its more chaotic to remove it later than to add it later